src/main/java/commands/AddCommand.java

```diff
@@ -0,0 +1,138 @@
+package commands;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.time.LocalDateTime;
+
+import javafx.scene.layout.VBox;
+import parser.Parser;
+import storage.Storage;
+import tasklist.TaskList;
+import ui.Ui;
+
+public class AddCommand extends Command {
+
+	enum TaskType {
+		TODO, DEADLINE, EVENT
+	}
+
+	private Task task;
+	private TaskType taskType;
+
+	public AddCommand(String[] messageArray) {
+		super(messageArray);
+		setTaskType(messageArray[0]);
+	}
+
+	public TaskType getTaskType() {
+		return this.taskType;
+	}
+
+	/**
+	 * Set task type from user input.
+	 * 
+	 * @param message
+	 */
+	private void setTaskType(String message) {
+		String commandLower = message.toLowerCase();
+		this.taskType = commandLower.startsWith("todo")
+				? TaskType.TODO
+				: commandLower.startsWith("deadline")
+						? TaskType.DEADLINE
+						: TaskType.EVENT;
+	}
+
+	/**
+	 * Set task from task type.
+	 * 
+	 * @param inputArr
+	 * @param ui
+	 * @param parser
+	 * @param storage
+	 * @param dialogContainer
+	 * @return boolean
+	 */

```

Comment:
I think it would be clearer if you could briefly explain what each parameter is for each JavaDoc documentation